bcrhp/src/bcrhp/schema/SiteClassificationSchema.ts

```diff
@@ -0,0 +1,58 @@
+import { z } from 'zod';
+
+const SiteClassificationSchema = z.object({
+    contributingResources: z
+        .string()
+        .min(1, { message: 'Number of Contributing Resources is required.' })
+        .max(250),
+    heritageClasses: z.array(z.string()).max(5),
+    functionCategory: z
+        .string()
+        .min(1, { message: 'Function Category is required.' })
+        .max(250),
+    heritageFunctions: z.array(z.string()).max(5),
+    heritageTheme: z
+        .string()
+        .min(1, { message: 'Heritage Theme is required.' })
+        .max(250),
+    heritageThemes: z.array(z.string()).max(5),
+});
+
+const requiredSiteClassificationSchema = SiteClassificationSchema.partial({});
+// @ts-ignore
+type SiteClassificationType = z.infer<typeof SiteClassificationSchema>;
+
+function getSiteClassificationSchema(): SiteClassificationType {
+    return new SiteClassification();
+}
+
+// @todo - Figure out object state - New/Updated/Deleted
+class SiteClassification implements SiteClassificationType {
+    constructor() {
+        this.contributingResources = '';
+        this.heritageClasses = [];
```

Comment:
I don' think the heritageClass is anywhere - Heritage Category is the value saved (Heritage Class is the grouping object, so maybe a sub-schema should have that name).